lib/search.rb

```diff
@@ -986,53 +986,30 @@ def posts_query(limit, type_filter: nil, aggregate_search: false)
         posts
       end
 
-    if aggregate_search
-      aggregate_relation = Post.unscoped
-        .select("subquery.topic_id id")
-        .group("subquery.topic_id")
-
-      posts = posts.select(posts.arel.projections)
-    end
-
     if @order == :latest
-      posts = posts.reorder("posts.created_at DESC")
-
       if aggregate_search
-        aggregate_relation = aggregate_relation
-          .select(
-            "MAX(subquery.post_number) post_number",
-            "MAX(subquery.created_at) created_at"
-          )
-          .order("created_at DESC")
+        posts = posts.order("MAX(posts.created_at) DESC")
```

Comment:
Why is that MAX function used here?

Reply:
If we're doing aggregation with `group`, we need to use an aggregation function in the ordering as well. In this case, the latest post in the topic is returned.